### DISCUSSION TITLE
Version 0.2.3

### DISCUSSION THREAD

**Discussion by m-haisham** [nacht-org/nacht#59] [Announcements]
### Added

- Check for new releases from github when app is opened
  - Download and prompt to install in android systems
- Created an abstraction layer over background tasks and notifications.
  - This should make it easier to implement any functionality that require them in the future.
- Added more interactivity to webview
  - Added url below title
  - Added 'Refresh' button
  - Added 'Open in browser' button
  - Added 'Share' button
- Added 'Clear history' button to delete all history in history tab
- Draw app below android system bottom navigation bar
  - This removes the black bar at bottom in android 13
- Added popup menu button to novel page
  - Added 'Refresh' button
  - Added 'Edit categories' button

### Changed

- Migrate auto_route to v7
- Selection AppBar is now always elevated
- Changed custom bottom bar elevation from 4 to 3
- Bluetooth is no longer recognized as a valid data connection
- Record chapter to reading history only when chapter is loaded
  - This prevents app from recording your attempt to read a chapter even though no content might have been loaded (ex: no internet connection)
- Switch to using default bottom sheet with handle bar
- Chapter list text in reader drawer is now smaller to show more chapters
- Clicking on 'In libary' button in novel page will now remove novel from all categories
- When popping state, change navigation index to `initialIndex` if `currentIndex` is not `initialIndex`
  - This is done to be more faithful to material 3 navigation
- Don't show default category in novel category picker dialog unless it is selected
- Open edit categories dialog in novel page on long press to 'In library'

### Fixed

- Correct name of android keystore file in workflow file
- Fixed novelpub.net novel scraping
- Fixed constraint error when deleting chapter
  - This is achieved by deleting all associated data when deleting chapter
- At end callback to load more in browse will now only be called if there is data #50
- Preload reader font so that there is no text jump when reader is initially opened
- Set reader status bar color according to reader theme (ex: dark on sepia even if app theme is dark)
- Don't allow 'pull-to-refresh' in updates page when selection is active
- Novel description can now be expanded or collapsed by clicking on the background behind description

<hr /><em>This discussion was created from the release <a href='https://github.com/nacht-org/nacht/releases/tag/v0.2.3'>Version 0.2.3</a>.</em>

**Comment by n4ff4h** [comment]
Lets goooooo

**Comment by Charlotte-br560** [comment]
Impressive updates! Your additions and improvements enhance usability and address several issues effectively. The incorporation of features like checking for new releases and refining the UI with additional buttons greatly enhances user experience. Your attention to detail in fixing issues such as correcting file names and resolving scraping problems is commendable. Keep up the great work!
